# Web Scraping


Tujuan dari project ini digunakan untuk mengetahui anomali daya minat dan daya beli masyarakat terhadap produk seblak pada platform Tokopedia agar dapat melakukan skema dropshipper produk seblak, yang pada akhirnya dapat menambah pandapatan. Oleh karena itu, perlu dilakukan analisis tentang penjualan seblak pada platform Tokopedia.



# Bagian A (Web Scraping)

In [1]:
from bs4 import BeautifulSoup
from selenium import webdriver
import pandas as pd
import time

Melakukan import library BeautifulSoup, webdriver, pandas, dan time

In [2]:
namaProdukList = []
hargaProdukList = []
penjualList = []
kotaTokoList = []
barangTerjualList = []
ratingProdukList = []



driver = webdriver.Chrome()

for page in range(1, 11):
    url= f"https://www.tokopedia.com/search?navsource=&page={page}&q=seblak&srp_component_id=02.01.00.00&srp_page_id=&srp_page_title=&st="
    driver.get(url)
    
    time.sleep(6)
    pageHeight = int(driver.execute_script("return document.body.scrollHeight"))
    for pixel in range(1, pageHeight, 10):
        driver.execute_script(f"window.scrollTo(0, {pixel});")
    
    html = driver.page_source
    soup = BeautifulSoup(html, "html.parser")

    datatestIdEl = soup.find("div", {"data-testid" : "divSRPContentProducts"})
    dataRowsEl = datatestIdEl.find_all("div", {"class" :"css-jza1fo"})

    for rowEl in dataRowsEl:
        ProdukEl = rowEl.find_all("div", {"class" : "css-llwpbs"})
   
        for produk in ProdukEl:

            # nama produk
            namaProdukEl = produk.find("div", {"class":"prd_link-product-name css-3um8ox"})
            
            if namaProdukEl is None:
                namaProdukList.append('No Data')
            else:
                namaProdukList.append(namaProdukEl.get_text())

            # harga produk
            hargaProdukEl = produk.find("div", {"class":"prd_link-product-price css-h66vau"})
            
            if hargaProdukEl is None:
                hargaProdukList.append('No Data')
            else:
                hargaProdukList.append(hargaProdukEl.get_text())

            # penjual
            penjualEl = produk.find("span", {"class":"prd_link-shop-name css-1kdc32b flip"})
            
            if penjualEl is None:
                penjualList.append('No Data')
            else:
                penjualList.append(penjualEl.get_text())

            # kota
            kotaTokoEl = produk.find("span", {"class":"prd_link-shop-loc css-1kdc32b flip"})
            
            if kotaTokoEl is None:
                kotaTokoList.append('No Data')
            else:
                kotaTokoList.append(kotaTokoEl.get_text())

            # barang terjual 
            barangTerjualEl = produk.find("span", {"class":"prd_label-integrity css-1sgek4h"})
            
            if barangTerjualEl is None:
                barangTerjualList.append('No Data')
            else:
                barangTerjualList.append(barangTerjualEl.get_text())

            # rating produk
            ratingProdukEl = produk.find("span", {"class":"prd_rating-average-text css-t70v7i"})
            
            if ratingProdukEl is None:
                ratingProdukList.append('No Data')
            else:
                ratingProdukList.append(ratingProdukEl.get_text())
    



Melakukan web scraping pada kolom pencarian seblak di platform Tokopedia. Adapun cara yang dilakukan yang pertama adalah membuat list untuk kolom Nama Produk, Harga Produk, Penjual, Kota Toko, Barang Terjual, dan Rating Produk. Setelah membuat list untuk kolom" yang sudah ditentukan, dalam hal ini membuat sebuah looping untuk setiap halaman pencarian seblak dari halaman 1 sampai dengan 10 dan juga melakukan looping untuk automasi scroll di tiap halaman pencarian seblak. Kemudian mencari setiap class pembungkus untuk mencari informasi tentang ama Produk, Harga Produk, Penjual, Kota Toko, Barang Terjual, dan Rating Produk yang akan dilooping dan pada akhirnya dimasukkan ke dalam list yang sebelumnya sudah dibuat.

In [3]:
df = pd.DataFrame({
    "Nama Produk" : namaProdukList,
    "Harga Produk" : hargaProdukList,
    "Penjual" : penjualList,
    "Kota Toko" : kotaTokoList,
    "Barang Terjual" : barangTerjualList,
    "Rating Produk" : ratingProdukList
})

df

Nama Produk Harga Produk  \
0                        Kylafood Seblak Original Play     Rp20.425   
1              Seblak Rafael, Seblak Coet Instan Halal     Rp25.000   
2                                        Seblak Instan      Rp8.000   
3    KERUPUK SEBLAK MENTAH 1KG WARNA WARNI - Mawar ...     Rp23.500   
4    Kerupuk Seblak Rafael Pedas / Seblak Mawar Ped...     Rp15.000   
..                                                 ...          ...   
832  Cuanki tahu bulat kering 50 pcs toping seblak ...     Rp18.000   
833                                 pedo seblak kencur     Rp89.000   
834  Seblak Original k.y.l Kylafood (Spesial Panen ...     Rp20.425   
835  siomay mini kering 250 gram toping seblak baso...      Rp8.000   
836  PAKET MIDYEAR SALE(Seblak Original, Bacil, Cir...     Rp76.000   

               Penjual        Kota Toko Barang Terjual Rating Produk  
0             kylafood          Bandung   1rb+ terjual           4.9  
1    Brother Meat Shop            Depok   250+ terjual           4.9  
2        ROEMAH MOSLEM    Kab. Boyolali    50+ terjual           5.0  
3           BociKakang  Jakarta Selatan   250+ terjual           5.0  
4      DUO BOCIL SNACK            Depok    80+ terjual           5.0  
..                 ...              ...            ...           ...  
832   CUANKI CULAMETAN       Kab. Garut    30+ terjual           4.8  
833          ANV_SNACK         Surabaya        No Data       No Data  
834           kylafood          Bandung   1rb+ terjual           4.9  
835   CUANKI CULAMETAN       Kab. Garut   100+ terjual           4.8  
836           kylafood          Bandung    40+ terjual           5.0  

[837 rows x 6 columns]

Dalam hal ini, list kolom yang sudah berisi data yang sudah diinginkan dimasukkan ke dalam dataframe dengan menggunakan pandas. Setelahnya memasukkan ke dalam data frame, melihat data-data di dalam dataframe yang sudah dibuat.

# Bagian B (Data Preparation)

In [4]:
df.head(5)

Nama Produk Harga Produk  \
0                      Kylafood Seblak Original Play     Rp20.425   
1            Seblak Rafael, Seblak Coet Instan Halal     Rp25.000   
2                                      Seblak Instan      Rp8.000   
3  KERUPUK SEBLAK MENTAH 1KG WARNA WARNI - Mawar ...     Rp23.500   
4  Kerupuk Seblak Rafael Pedas / Seblak Mawar Ped...     Rp15.000   

             Penjual        Kota Toko Barang Terjual Rating Produk  
0           kylafood          Bandung   1rb+ terjual           4.9  
1  Brother Meat Shop            Depok   250+ terjual           4.9  
2      ROEMAH MOSLEM    Kab. Boyolali    50+ terjual           5.0  
3         BociKakang  Jakarta Selatan   250+ terjual           5.0  
4    DUO BOCIL SNACK            Depok    80+ terjual           5.0

Dalam hal ini, melakukan penampilan 5 baris data teratas yang ada di dalam dataframe. Kemudian jika dilihat dari 5 baris teratas, dapat disimpulkan terdapat berbagai macam nama produk seblak yang dijual di platform tokopedia. Pada kelima produk tersebut terdapat perbedaan harga yang ditetapkan pada platform tokopedia. Produk seblak juga dijual oleh berbagai macam penjual denga kota yang berbeda-beda seperti Toko kylafood yang berdomisili di Bandung atau Non-Jabodetak. Lebih lanjut, dapat dilihat juga bahwa setiap produk seblak memiliki perbedaan barang terjual dan rating produk di Tokopedia. 

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 837 entries, 0 to 836
Data columns (total 6 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   Nama Produk     837 non-null    object
 1   Harga Produk    837 non-null    object
 2   Penjual         837 non-null    object
 3   Kota Toko       837 non-null    object
 4   Barang Terjual  837 non-null    object
 5   Rating Produk   837 non-null    object
dtypes: object(6)
memory usage: 39.4+ KB


Berdasarkan data diatas, dapat disimpulkan bahwa terdapat 826 baris yang terdapat di dalam dataframe yang terdiri dari 6 kolom utama, yaitu kolom Nama Produk, Harga Produk, Penjual, Kota Toko, Barang Terjual, dan Rating Produk. Adapun tipe data pada setiap kolom masih bertipe object atau string. Oleh karena itu, perlu dilakukan perubahan tipe data object menjadi tipe data integer atau float tergantung value yang ada di dalam kolomnya, seperti tipe data integer untuk harga produk dan tipe data float untuk rating produk. Adapun total baris yang didapat sebanyak 837 data.

In [6]:
missing_values = df.apply(lambda col: col == 'No Data')

jumlahMissingValue = missing_values.sum()

print(jumlahMissingValue)

Nama Produk        0
Harga Produk       0
Penjual            1
Kota Toko          1
Barang Terjual    58
Rating Produk     77
dtype: int64


Dalam hal ini menggunakan fungsi Berdasarkan data diatas, terdapat missing value "No Data" pada kolom Barang Terjual, Penjual, Kota Toko, dan Rating Produk di dataframe. Oleh karena itu, dibutuhkan treatment pada missing value tersebut agar tidak membuat data outlier menjadi lebih besar. Dalam hal ini treatment yang dilakukana adalah melakukan cleaning data dengan cara menghapus baris yang memiliki value "No Data" pada dataframe.

In [7]:
df['Harga Produk'] = df['Harga Produk'].str.replace('Rp', '')

df['Harga Produk'] = df['Harga Produk'].str.replace('.', '')

df['Barang Terjual'] = df['Barang Terjual'].str.replace('terjual', '')

df['Barang Terjual'] = df['Barang Terjual'].str.replace('+', '')

df['Barang Terjual'] = df['Barang Terjual'].str.replace('rb', '000')


Dalam proses diatas, dilakukan pembersihan data pada kolom Harga Produk dan Barang terjual. Pada kolom Harga Produk dilakukan cleaning data seperti mengganti atau menghapus simbol Rp dan tanda titik pada value. Kemudian pada kolom Barang terjual dilakukan cleaning data seperti membuang data terjual dan tanda '+' pada value, dan mengganti kata rb menjadi 000 pada value agar jika dilakukan perubahan tipe data menjadi string tetap menjaga value asli dari web scraping.

In [8]:
df_clean = df[(df['Barang Terjual'] != "No Data") 
              & (df['Harga Produk'] != "No Data") 
              & (df['Rating Produk'] != "No Data") 
              & (df['Nama Produk'] != "No Data")
              & (df['Penjual'] != "No Data")
              & (df['Kota Toko'] != "No Data")]

df_clean.info()

<class 'pandas.core.frame.DataFrame'>
Index: 759 entries, 0 to 836
Data columns (total 6 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   Nama Produk     759 non-null    object
 1   Harga Produk    759 non-null    object
 2   Penjual         759 non-null    object
 3   Kota Toko       759 non-null    object
 4   Barang Terjual  759 non-null    object
 5   Rating Produk   759 non-null    object
dtypes: object(6)
memory usage: 41.5+ KB


Dalam proses diatas dilakukan untuk menghapus missing value ('No Data') pada setiap kolom di dataframe. Kemudian proses tersebut dimasukkan ke dalam variabel df_clean agar data yang sudah dibersihkan disimpan di dalam variabel tersebut. Kemudian jika dilihat dari summary data, dapat dilihat bahwa terdapat pengurangan jumlah baris sebesar 759 baris pada data frame.

In [9]:
df_clean['Harga Produk'] = df_clean['Harga Produk'].astype(int)
df_clean['Barang Terjual'] = df_clean['Barang Terjual'].astype(int)
df_clean['Rating Produk'] = df_clean['Rating Produk'].astype(float)

df_clean.info()

<class 'pandas.core.frame.DataFrame'>
Index: 759 entries, 0 to 836
Data columns (total 6 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Nama Produk     759 non-null    object 
 1   Harga Produk    759 non-null    int32  
 2   Penjual         759 non-null    object 
 3   Kota Toko       759 non-null    object 
 4   Barang Terjual  759 non-null    int32  
 5   Rating Produk   759 non-null    float64
dtypes: float64(1), int32(2), object(3)
memory usage: 35.6+ KB


C:\Users\Kelvin Rizky\AppData\Local\Temp\ipykernel_22356\2876615595.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_clean['Harga Produk'] = df_clean['Harga Produk'].astype(int)
C:\Users\Kelvin Rizky\AppData\Local\Temp\ipykernel_22356\2876615595.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_clean['Barang Terjual'] = df_clean['Barang Terjual'].astype(int)
C:\Users\Kelvin Rizky\AppData\Local\Temp\ipykernel_22356\2876615595.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of 

Dalam proses ini, dilakukan perubahan tipe data pada kolom Harga Produk dan Barang Terjual, dari tipe data string menjadi integer pada df_clean, dan perubahan tipe data pada kolom Rating Produk dari tipe data string menjadi float. Kemudian, jika dilihat dari summary data, perubahan tipe data yang diinginkan berhasil dilakukan.

In [10]:
df_clean

Nama Produk  Harga Produk  \
0                        Kylafood Seblak Original Play         20425   
1              Seblak Rafael, Seblak Coet Instan Halal         25000   
2                                        Seblak Instan          8000   
3    KERUPUK SEBLAK MENTAH 1KG WARNA WARNI - Mawar ...         23500   
4    Kerupuk Seblak Rafael Pedas / Seblak Mawar Ped...         15000   
..                                                 ...           ...   
831                        Seblak Viral Odjolali Snack         14900   
832  Cuanki tahu bulat kering 50 pcs toping seblak ...         18000   
834  Seblak Original k.y.l Kylafood (Spesial Panen ...         20425   
835  siomay mini kering 250 gram toping seblak baso...          8000   
836  PAKET MIDYEAR SALE(Seblak Original, Bacil, Cir...         76000   

               Penjual        Kota Toko  Barang Terjual  Rating Produk  
0             kylafood          Bandung            1000            4.9  
1    Brother Meat Shop            Depok             250            4.9  
2        ROEMAH MOSLEM    Kab. Boyolali              50            5.0  
3           BociKakang  Jakarta Selatan             250            5.0  
4      DUO BOCIL SNACK            Depok              80            5.0  
..                 ...              ...             ...            ...  
831     Odjolali Snack          Bandung              25            5.0  
832   CUANKI CULAMETAN       Kab. Garut              30            4.8  
834           kylafood          Bandung            1000            4.9  
835   CUANKI CULAMETAN       Kab. Garut             100            4.8  
836           kylafood          Bandung              40            5.0  

[759 rows x 6 columns]

Melakukan penampilan dataframe yang sudah dibersihkan untuk memastikan bahwa kolom yang berisikan tipe data number sudah sesuai dengan value asli pada platform Tokopedia.

# Bagian C (Business Understanding/Problem Statement)

- Specific: 
. Mengetahui anomali dari daya minat dan daya beli seblak di kalangan masyarakat pada platfrom Tokopedia untuk merumuskan strategi bisnis dropshipper.

- Measurable: 
. Mendapatkan informasi anomali data minat dan daya beli seblak pada platform Tokopedia berdasarkan hasil analisis dari harga produk, banyak barang terjual, rating produk, dan potensi pendapatan agar mendukung pengambilan keputusan strategi bisnis dropshipper.


- Achievable:
. Menggunakan webscraping untuk mendapatkan informasi data tentang seblak pada platform Tokopedia.
. Memperoleh nilai rata-rata dari harga produk, barang terjual, dan rating produk dari produk seblak.
. Mengestimasi potensi pendapatan jika menerapkan strategi bisnis dropshipper.
. Memperoleh informasi apakah terdapat perbedaan harga barang di Jabodetabek dengan non-Jabodetabek.
. Memperoleh informasi apakah terdapat kecenderungan seseorang dalam membeli produk seblak ketika harganya murah.

- Relevant:
. Berdasarkan olahan dan analisis informasi anomali data minat dan daya beli seblak pada platform Tokopedia, jika adanya anomali daya minat dan daya beli masyarakat yang tinggi terhadap produk seblak, maka strategi dropshipper dapat diterapkan dan selanjutnya dapat memilih toko seblak yang memiliki rating baik serta barang terjual yang besar.


- Time-bound:
. Mengumpulkan informasi data seblak pada platform Tokopedia dengan menggunakan webscraping selama 3 hari.
. Melakukan data preparation terhadap informasi data yang didapatkan dari webscraping selama 2 hari.
. Melakukan analisa data selama 1 hari.


Problem Statement :
. Mengidentifikasi anomali dalam minat dan daya beli masyarakat terhadap produk seblak di platform Tokopedia melalui analisis harga produk, jumlah barang terjual, rating produk, dan potensi pendapatan dalam waktu 6 hari, untuk merumuskan strategi bisnis dropshipper yang efektif.


# Bagian D

## Bagian 1

In [11]:
# menghitung mean, median, dan modus dari harga produk.

mean_harga = df_clean["Harga Produk"].mean()
median_harga = df_clean["Harga Produk"].median(axis=0)
median_harga = df_clean["Harga Produk"].quantile(0.5)
std_harga = df_clean["Harga Produk"].std()
skewness_harga = df_clean["Harga Produk"].skew()
kurtosis_harga = df_clean["Harga Produk"].kurt()

print(f'Mean harga adalah {mean_harga}')
print(f'Median harga adalah {median_harga}')
print(f'Standar Deviasi harga  adalah {std_harga}')
print(f'Skewness harga adalah {skewness_harga}')
print(f'Kurtosis harga  adalah {kurtosis_harga}')

Mean harga adalah 23126.084321475624
Median harga adalah 15000.0
Standar Deviasi harga  adalah 25721.966375552805
Skewness harga adalah 3.442962156224897
Kurtosis harga  adalah 16.74346052063465


Berdasarkan data harga produk yang dianalisis, dapat dilihat bahwa terdapat mean sebesar 23126.084321475624 dan median sebesar 15000, yang berarti terdapat kemungkinan data yang condong ke kanan atau negative skewness. Kemudian dapat dilihat juga bahwa terdapat standar deviasi sebesar 25721.966375552805, yang dapat dimaknai bawha sebaran data harga cukup tinggi atau jauh dari meannya. Lebih lanjut dari data skewness sebesar 3.442962156224897 dapat disimpulkan bawha sebaran data higly skewed karena data skewed harga lebih dari 1. Dari kurtosis harga sebesar 16.74346052063465 dapat dimaknai bahwa sebaran datanya adalah Leptokurtic distribution karena data kurtosisnya lebih dari 0. Berdasarkan analisis data-data tersebut dapat disimpulkan bahwa terdapat data outlier pada kolom harga produk.

In [12]:
# menghitung mean, median, dan modus dari barang terjual.

mean_barang_terjual = df_clean["Barang Terjual"].mean()
median_barang_terjual  = df_clean["Barang Terjual"].median(axis=0)
median_barang_terjual  = df_clean["Barang Terjual"].quantile(0.5)
std_barang_terjual = df_clean["Barang Terjual"].std()
skewness_barang_terjual = df_clean["Barang Terjual"].skew()
kurtosis_barang_terjual = df_clean["Barang Terjual"].kurt()


print(f'Mean barang terjual adalah {mean_barang_terjual}')
print(f'Median barang terjual adalah {median_barang_terjual}')
print(f'Standar Deviasi barang terjual adalah {std_barang_terjual}')
print(f'Skewness barang terjual adalah {skewness_barang_terjual}')
print(f'Kurtosis barang terjual adalah {kurtosis_barang_terjual}')

Mean barang terjual adalah 414.63899868247694
Median barang terjual adalah 100.0
Standar Deviasi barang terjual adalah 1122.9476695067824
Skewness barang terjual adalah 5.8561920507935294
Kurtosis barang terjual adalah 40.68052824982278


Berdasarkan data barang terjual yang dianalisis, dapat dilihat bahwa terdapat mean sebesar 414.63899868247694 dan median sebesar 100.0, yang berarti terdapat kemungkinan data yang condong ke kanan atau negative skewness. Kemudian dapat dilihat juga bahwa terdapat standar deviasi sebesar 1122.9476695067824, yang dapat dimaknai bawha sebaran data harga cukup tinggi atau jauh dari meannya. Lebih lanjut dari data skewness sebesar 5.8561920507935294 dapat disimpulkan bawha sebaran data higly skewed karena data skewed barang terjual lebih dari 1. Dari kurtosis barang terjual sebesar 40.68052824982278 dapat dimaknai bahwa sebaran datanya adalah Leptokurtic distribution karena data kurtosisnya lebih dari 0. Berdasarkan analisis data-data tersebut dapat disimpulkan bahwa terdapat data outlier pada kolom barang terjual.

In [13]:
# menghitung mean, median, dan modus dari rating produk.

mean_rating = df_clean["Rating Produk"].mean()
median_rating = df_clean["Rating Produk"].median(axis=0)
median_rating = df_clean["Rating Produk"].quantile(0.5)
std_rating = df_clean["Rating Produk"].std()
skewness_rating = df_clean["Rating Produk"].skew()
kurtosis_rating = df_clean["Rating Produk"].kurt()


print(f'Mean rating adalah {mean_rating}')
print(f'Median rating adalah {median_rating}')
print(f'Standar Deviasi rating adalah {std_rating}')
print(f'Skewness rating adalah {skewness_rating}')
print(f'Kurtosis rating adalah {kurtosis_rating}')

Mean rating adalah 4.874176548089592
Median rating adalah 4.9
Standar Deviasi rating adalah 0.14822547692164478
Skewness rating adalah -2.004348385652977
Kurtosis rating adalah 5.9841697491120165


Berdasarkan data rating produk yang dianalisis, dapat dilihat bahwa terdapat mean sebesar 4.874176548089592 dan median sebesar 4.9, yang berarti terdapat kemungkinan data yang condong ke kiri atau positif skewness. Kemudian dapat dilihat juga bahwa terdapat standar deviasi sebesar 0.14822547692164478, yang dapat dimaknai bawha sebaran data harga cukup tinggi atau jauh dari meannya. Lebih lanjut dari data skewness sebesar -2.004348385652977 dapat disimpulkan bawha sebaran data higly skewed karena data skewed rating produk lebih dari 1. Dari kurtosis rating produk sebesar 5.9841697491120165 dapat dimaknai bahwa sebaran datanya adalah Leptokurtic distribution karena data kurtosisnya lebih dari 0. Berdasarkan analisis data-data tersebut dapat disimpulkan bahwa terdapat data outlier pada kolom rating produk.

## Bagian 2

In [14]:
df_clean["Pendapatan"] = df_clean["Harga Produk"] * df_clean["Barang Terjual"]

C:\Users\Kelvin Rizky\AppData\Local\Temp\ipykernel_22356\1654585966.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_clean["Pendapatan"] = df_clean["Harga Produk"] * df_clean["Barang Terjual"]


Dalam hal ini melakukan pembentukan kolom baru dengan nama pendapatan yang mana isi value pada kolom pendapatan didapatkan dengan mengkalikan value harga produk dengan barang terjual. Pembentukan kolom tersebut untuk melihat potensi maksimum dan minimum pendapat jika berjualan seblak.

In [15]:
df_clean.info()

<class 'pandas.core.frame.DataFrame'>
Index: 759 entries, 0 to 836
Data columns (total 7 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Nama Produk     759 non-null    object 
 1   Harga Produk    759 non-null    int32  
 2   Penjual         759 non-null    object 
 3   Kota Toko       759 non-null    object 
 4   Barang Terjual  759 non-null    int32  
 5   Rating Produk   759 non-null    float64
 6   Pendapatan      759 non-null    int32  
dtypes: float64(1), int32(3), object(3)
memory usage: 38.5+ KB


Menampilkan summary data setelah menambahkan kolom pendapatan.

In [16]:
from scipy import stats
import numpy as np

std = df_clean["Pendapatan"].std()
N = len(df_clean)
low, up = stats.norm.interval(0.95,loc=df_clean["Pendapatan"].mean(),scale=std/np.sqrt(N))
print('Lower Limit Pendapatan:',low)
print('Upper Limit Pendapatan:',up)


Lower Limit Pendapatan: 5173215.5386430025
Upper Limit Pendapatan: 7931605.57861655


Berdasarkan syxtax diatas yang menggunakan confidence level 95%, dapat disimpulkan bahwa potensi maksimum pendapatan jika berjualan seblak sebesar 7931605.57861655. Kemudian potensi minimum pendapatan jika berjualan seblak sebesar 5173215.5386430025.

## Bagian 3

In [17]:
df_clean["Kota Toko"].unique()

array(['Bandung', 'Depok', 'Kab. Boyolali', 'Jakarta Selatan',
       'Tangerang Selatan', 'Jakarta Barat', 'Jakarta Timur',
       'Kab. Bandung', 'Jakarta Pusat', 'Kab. Bekasi', 'Tasikmalaya',
       'Kab. Garut', 'Kab. Indramayu', 'Surabaya', 'Kab. Bogor',
       'Tangerang', 'Kab. Tangerang', 'Bekasi', 'Kab. Sidoarjo',
       'Kab. Sumedang', 'Semarang', 'Cimahi', 'Kab. Majalengka', 'Medan',
       'Banjarbaru', 'Makassar', 'Pekanbaru', 'Jakarta Utara',
       'Palembang', 'Kab. Sukabumi', 'Kab. Bandung Barat', 'Surakarta',
       'Malang', 'Kab. Bantul', 'Kab. Cianjur', 'Kab. Mojokerto',
       'Kab. Demak', 'Kab. Jember', 'Kab. Purwakarta', 'Kab.Ciamis',
       'Kab. Malang', 'Batam', 'Kab. Tasikmalaya', 'Bogor', 'Yogyakarta',
       'Sukabumi', 'Kab. Klaten', 'Serang', 'Kab. Jombang', 'Kab. Gresik',
       'Kab. Lebak'], dtype=object)

Menampilkan nama-nama kota yang ada di dalam kolom Kota Toko untuk dilakukan looping yang utamanya memisahkan kota Jabodetabek dan Non Jabodetabek.

In [18]:
Domisili = []

for kota in df_clean["Kota Toko"]:
  if kota == 'Tangerang Selatan':
    Domisili.append('Jabodetabek')
  elif kota == 'Jakarta Selatan':
    Domisili.append('Jabodetabek')
  elif kota == 'Depok':
    Domisili.append('Jabodetabek')
  elif kota == 'Jakarta Barat':
    Domisili.append('Jabodetabek')
  elif kota == 'Kab. Bogor':
    Domisili.append('Jabodetabek')
  elif kota == 'Kab. Tangerang':
    Domisili.append('Jabodetabek')
  elif kota == 'Jakarta Timur':
    Domisili.append('Jabodetabek')
  elif kota == 'Jakarta Utara':
    Domisili.append('Jabodetabek')
  elif kota == 'Bekasi':
    Domisili.append('Jabodetabek')
  elif kota == 'Kab. Bekasi':
    Domisili.append('Jabodetabek')
  elif kota == 'Tangerang':
    Domisili.append('Jabodetabek')
  elif kota == 'Jakarta Pusat':
    Domisili.append('Jabodetabek')
  elif kota == 'Bogor':
    Domisili.append('Jabodetabek')  
  else:
    Domisili.append('Non Jabodetabek')

df_clean['Domisili'] = Domisili

C:\Users\Kelvin Rizky\AppData\Local\Temp\ipykernel_22356\2464988978.py:33: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_clean['Domisili'] = Domisili


Pada proses diatas melakukan proses looping dimana jika terdapat kota yang ada di dalam Jabodetabek, maka kota tersebut masuk ke dalam kategori Jabodetabek dan kota yang ada di luar Jabodetabek, maka kota tersebut masuk ke dalam kategori Non-Jabodetabek. Kemudian data-data tersebut dimasukkan ke dalam list Domisili dan menambahkan kolom domisili pada dataframe.

In [19]:
df_clean

Nama Produk  Harga Produk  \
0                        Kylafood Seblak Original Play         20425   
1              Seblak Rafael, Seblak Coet Instan Halal         25000   
2                                        Seblak Instan          8000   
3    KERUPUK SEBLAK MENTAH 1KG WARNA WARNI - Mawar ...         23500   
4    Kerupuk Seblak Rafael Pedas / Seblak Mawar Ped...         15000   
..                                                 ...           ...   
831                        Seblak Viral Odjolali Snack         14900   
832  Cuanki tahu bulat kering 50 pcs toping seblak ...         18000   
834  Seblak Original k.y.l Kylafood (Spesial Panen ...         20425   
835  siomay mini kering 250 gram toping seblak baso...          8000   
836  PAKET MIDYEAR SALE(Seblak Original, Bacil, Cir...         76000   

               Penjual        Kota Toko  Barang Terjual  Rating Produk  \
0             kylafood          Bandung            1000            4.9   
1    Brother Meat Shop            Depok             250            4.9   
2        ROEMAH MOSLEM    Kab. Boyolali              50            5.0   
3           BociKakang  Jakarta Selatan             250            5.0   
4      DUO BOCIL SNACK            Depok              80            5.0   
..                 ...              ...             ...            ...   
831     Odjolali Snack          Bandung              25            5.0   
832   CUANKI CULAMETAN       Kab. Garut              30            4.8   
834           kylafood          Bandung            1000            4.9   
835   CUANKI CULAMETAN       Kab. Garut             100            4.8   
836           kylafood          Bandung              40            5.0   

     Pendapatan         Domisili  
0      20425000  Non Jabodetabek  
1       6250000      Jabodetabek  
2        400000  Non Jabodetabek  
3       5875000      Jabodetabek  
4       1200000      Jabodetabek  
..          ...              ...  
831      372500  Non Jabodetabek  
832      540000  Non Jabodetabek  
834    20425000  Non Jabodetabek  
835      800000  Non Jabodetabek  
836     3040000  Non Jabodetabek  

[759 rows x 8 columns]

Melakukan penampilan dataframe untuk melakukan pengecekan kolom domisili.

In [20]:
Jabodetabek = df_clean[df_clean['Domisili']=='Jabodetabek']
nonJabodetabek = df_clean[df_clean['Domisili']=='Non Jabodetabek']

Dalam proses diatas melakukan pembentukan variabel Jabodetabek, yang mana value pada variabel Jabodetabek adalah value 'Jabodetabek' yang ada di dalam kolom domisili dan melakukan pembentukan variabel nonJabodetabek, yang mana value pada variabel nonJabodetabek adalah value 'Non Jabodetabek' yang ada di dalam kolom domisili untuk dapat menganalisis perbedaan kota yang ada Jabodetabek dan Non Jabodetabek dari sisi harga produk.

Oleh karena itu dibutuhkan sebuah Hipotesis null dan Hipotesis alternatif untuk menjawab apakah ada perbedaan harga produk Jabodetabek dan Non Jabodetabek.

H0 : Harga Produk Jabodetabek == Harga Produk Non Jabodetabek

H1 : Harga Produk Jabodetabek != Harga Produk Non Jabodetabek

In [21]:
t_stat, p_val = stats.ttest_ind(Jabodetabek['Harga Produk'],nonJabodetabek['Rating Produk'])
print('T-Statistic:',t_stat)
print('P-value:',p_val)

T-Statistic: 17.60032205726366
P-value: 2.1986974973095883e-58


Berdasarkan analisis P-value diatas, dapat dianalisis bahwa nilai P-value (2.1986974973095883e-58) < critical value (0.05) yang dapat diartikan bahwa terdapat perbedaan harga barang antara kota di Jabodetabek dan Non Jabodetabek (Tolak  H0).

## Bagian 4

In [22]:
from scipy import stats

corr_rho, pval_s = stats.spearmanr(df_clean['Harga Produk'], df_clean['Rating Produk'])
print(f"rho-correlation: {corr_rho:.2f}, p-value: {pval_s}")


rho-correlation: 0.12, p-value: 0.001131112396701227


H0 = Ada korelasi yang nyata antara harga produk dengan rating produk

H1 = Tidak ada korelasi yang nyata antara harga produk dengan rating produk

Berdasarkan analisis P-value diatas, didapatkan bahwa nilai korelasi yang didapat 0.12 atau korelasi positif yang lemah. Korelasi positif lemah dapat dimaknai karena nilainya lebih mendekati 0 daripada 1. Artinya, dapat disimpulkan bahwa terdapat hubungan positif yang lemah antara harga produk dan rating produk, jadi ketika harga makin tinggi maka rating produknya akan semakin tinggi juga. Oleh karena itu tidak ada hubungan antara harga rendah atau murah dengan preferensi membeli. Lebih lanjut, dapat dianalisis juga bahwa nilai P-value (0.001131112396701227) < critical value (0.05) yang dapat diartikan bahwa tidak ada korelasi yang nyata antara harga produk dengan rating produk. Dengan demikian dapat disimpulkan bawha tidak ada kecenderungan preferensi membeli dengan harga yang murah.

# Bagian E (Kesimpulan)

Kesimpulan yang dapat diambil dari hasil analisis diatas adalah : 

1. Berdasarkan hasil analisis, dapat disimpulkan bahwa untuk strategi dropshipper seblak di platform Tokopedia, sebaiknya memilih produk dengan harga di bawah atau pada rata-rata (23126.08) untuk memenuhi preferensi beli konsumen, serta fokus pada produk dengan barang terjual di atas rata-rata (414.63) dan rating di atas rata-rata (4.87) untuk meningkatkan kepercayaan konsumen.

2. Dari segi pendapatan, strategi dropshipper memiliki potensi pendapatan antara Rp5juta hingga Rp7juta, yang mana memberikan gambaran mengenai potensi keuntungan yang dapat diperoleh dari skema bisnis ini.

3. Analisis menunjukkan bahwa terdapat perbedaan harga produk seblak di dalam dan di luar Jabodetabek, sehingga pemilihan penjual seblak untuk dropshipper sebaiknya mempertimbangkan lokasi.

4. Analisis menunjukkan bahwa daya beli masyarakat (preferensi suka) terhadap produk seblak tidak dipengaruhi oleh harga yang murah, menunjukkan bahwa faktor harga tidak menjadi penentu utama dalam preferensi pembelian konsumen.
